src/components/common/Layout/components/Account/components/Notifications/components/web3Inbox.ts

```diff
@@ -0,0 +1,104 @@
+import { useToast } from "@chakra-ui/react"
+import {
+  initWeb3InboxClient,
+  useNotifications,
+  usePrepareRegistration,
+  useRegister,
+  useSubscribe,
+  useSubscription,
+  useWeb3InboxAccount,
+} from "@web3inbox/react"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import { useEffect, useState } from "react"
+import { useAccount, useSignMessage } from "wagmi"
+
+export const WEB3_INBOX_INIT_PARAMS = {
+  projectId: process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID,
+  domain: "guild.xyz",
+  allApps: process.env.NODE_ENV !== "production",
+}
+
+export const useWeb3InboxSubscription = () => {
+  const [isSigningWeb3Inbox, setIsSigningWeb3Inbox] = useState(false)
+
+  useEffect(() => {
+    initWeb3InboxClient(WEB3_INBOX_INIT_PARAMS)
+  }, [])
+
+  const toast = useToast()
+  const showErrorToast = useShowErrorToast()
+  const { signMessageAsync } = useSignMessage()
+
+  const { prepareRegistration, isLoading: isPreparing } = usePrepareRegistration()
+  const { register, isLoading: isRegisteringWeb3Inbox } = useRegister()
+  const { address, isConnecting, isReconnecting, status } = useAccount()
+  const {
+    data: w3IAccount,
```

Comment:
I think it's safe to call this `account`, we're inside the `useWeb3InboxMessaging` hook, so we know that this is the user's Web3Inbox account